modules/nf-scil/registration/tractogram/main.nf

```diff
@@ -0,0 +1,83 @@
+process REGISTRATION_TRACTOGRAM {
+    tag "$meta.id"
+    label 'process_single'
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'scil.usherbrooke.ca/containers/scilus_1.6.0.sif':
+        'scilus/scilus:1.6.0' }"
+
+    input:
+    tuple val(meta), path(anat), path(transfo), path(centroids_dir), path(ref) /* optional, value = [] */
+
+    output:
+    tuple val(meta), path("*__*.trk"), emit: warped_bundle
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def reference = "$ref" ? "--reference $ref" : ""
```

Comment:
I'm not sure you need $ for the first ref : 
def reference = ref ? "--reference $ref" : ""